pytorch_blade/src/torch-mlir/lib/Dialect/TorchConversion/Transforms/DiscLowerOperatorOp.cpp

```diff
@@ -0,0 +1,160 @@
+// limitations under the License.
+
+#include "mlir/Dialect/Arithmetic/IR/Arithmetic.h"
+#include "mlir/Dialect/ControlFlow/IR/ControlFlowOps.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/Math/IR/Math.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"
+#include "mlir/IR/Builders.h"
+#include "mlir/Transforms/DialectConversion.h"
+#include "stablehlo/dialect/ChloOps.h"
+#include "tensorflow/compiler/mlir/disc/IR/hlo_disc_ops.h"
+#include "torch-mlir/Conversion/MhloPasses.h"
+#include "torch-mlir/Conversion/TorchToMhlo/TorchToMhlo.h"
+#include "torch-mlir/Dialect/Torch/IR/TorchDialect.h"
+#include "torch-mlir/Dialect/Torch/IR/TorchOps.h"
+#include "torch-mlir/Dialect/Torch/IR/TorchTypes.h"
+#include "torch-mlir/Dialect/TorchConversion/Transforms/BackendTypeConversion.h"
+
+using namespace mlir;
+using namespace mlir::torch;
+using namespace mlir::torch::Torch;
+using namespace mlir::torch::TorchConversion;
+
+namespace {
+
+class ConvertOperatorOp : public OpConversionPattern<OperatorOp> {
+ public:
+  using OpConversionPattern<OperatorOp>::OpConversionPattern;
+  using OpAdaptor = typename OperatorOp::Adaptor;
+  LogicalResult matchAndRewrite(
+      OperatorOp op,
+      OpAdaptor adaptor,
+      ConversionPatternRewriter& rewriter) const override {
+    Location loc = op.getLoc();
+    auto name = op.name();
+    if ("torch_blade.fake_quant" == name) {
+      auto operands = adaptor.operands();
+      auto input = operands[0];
+      auto scale = operands[1];
+      auto zeroPoint = operands[2];
+
+#define I64_VAR_FROM_CONST_OPERAND(var, idx)                         \
+  int64_t var;                                                       \
+  if (!matchPattern(op.operands()[idx], m_TorchConstantInt(&var))) { \
+    return op.emitError(#var " must be a Scalar constant int");      \
+  }
+      I64_VAR_FROM_CONST_OPERAND(qmin, 3);
+      I64_VAR_FROM_CONST_OPERAND(qmax, 4);
+      I64_VAR_FROM_CONST_OPERAND(numBits, 5);
+#undef I64_VAR_FROM_CONST_OPERAND
+
+      SmallVector<int64_t, 4> axis;
+      if (!matchPattern(op.operands()[6], m_TorchConstantIntList(axis))) {
+        return op.emitError("Only constant dims are currently supported");
+      }
+
+#define BOOL_VAR_FROM_CONST_OPERAND(var, idx)                         \
+  bool var;                                                           \
+  if (!matchPattern(op.operands()[idx], m_TorchConstantBool(&var))) { \
+    return op.emitError(#var " must be a Scalar constant boolean");   \
+  }
+
+      BOOL_VAR_FROM_CONST_OPERAND(useSigned, 7);
+      BOOL_VAR_FROM_CONST_OPERAND(useSymmetric, 8);
+      BOOL_VAR_FROM_CONST_OPERAND(useDynamic, 9);
+      BOOL_VAR_FROM_CONST_OPERAND(usePerChannel, 10);
+#undef BOOL_VAR_FROM_CONST_OPERAND
+
+      auto resultTy = input.getType().dyn_cast<RankedTensorType>();
+      if (!resultTy) {
+        return op.emitError("failed to get type of input");
+      }
+      auto castedResultTy =
```

Comment:
what's the supposed element type of `input`?  Why explicit cast to f32?

Reply:
updated to check if input is f32 and use input type as output's.